test/cmd/node-management.sh

```diff
@@ -72,15 +72,16 @@ __EOF__
 __EOF__
 
   # taint/untaint
-  # Pre-condition: node has no taints
-  kube::test::get_object_assert "nodes 127.0.0.1" "{{.spec.taints}}" '<no value>'
+  # Pre-condition: due to #67734, the fake Node 127.0.0.1 is applied with 
+  # a node.kubernetes.io/unreachable:NoSchedule taint
+  kube::test::get_object_assert "nodes 127.0.0.1" '{{range .spec.taints}}{{.key}}:{{.effect}} {{end}}' "node.kubernetes.io/unreachable:NoSchedule "
   # taint can add a taint
   kubectl taint node 127.0.0.1 dedicated=foo:PreferNoSchedule
-  kube::test::get_object_assert "nodes 127.0.0.1" '{{range .spec.taints}}{{.effect}}{{end}}' 'PreferNoSchedule'
+  kube::test::get_object_assert "nodes 127.0.0.1" '{{range .spec.taints}}{{.key}}:{{.effect}} {{end}}' "dedicated:PreferNoSchedule node.kubernetes.io/unreachable:NoSchedule "
   # taint can remove a taint
   kubectl taint node 127.0.0.1 dedicated-
-  # Post-condition: node has no taints
-  kube::test::get_object_assert "nodes 127.0.0.1" "{{.spec.taints}}" '<no value>'
+  # Post-condition: node has a node.kubernetes.io/unreachable:NoSchedule taint
+  kube::test::get_object_assert "nodes 127.0.0.1" '{{range .spec.taints}}{{.key}}:{{.effect}} {{end}}' "node.kubernetes.io/unreachable:NoSchedule "
```

Comment:
likewise, can we check for the absence of the `dedicated:PreferNoSchedule` taint, rather than explicitly listing all other present taints?

Reply:
It's possible, but just current kube::test::get_object_assert implementation is fixed to a pattern to matching ^<expected string>$. Let me try to work it around.

Reply:
Resolved using go-template trick.